base/broadcast.jl

```diff
@@ -1244,22 +1225,21 @@ macro __dot__(x)
     esc(__dot__(x))
 end
 
-function Base.show(io::IO, bc::Broadcasted)
-    print(io, "Broadcasted(", bc.f)
-    args = bc.args
-    while args != TupleLLEnd()
-        print(io, ", ", args.head)
-        args = args.rest
-    end
-    print(io, ')')
-end
+Base.show(io::IO, bc::Broadcasted) = print(io, "Broadcasted(", bc.f, ", ", bc.args, ')')
 
-function make_kwsyntax(f, args...; kwargs...)
-    args′ = make_TupleLL(args...)
-    g = (args...)->f(args...; kwargs...)
-    return Broadcasted(g, args′)
+struct TypeArgFunction{F, T, n} <: Function
+    f::F
 end
```

Comment:
`Type{T}` isn't a unique object. Maybe this doesn't matter for our use cases, but it might be better to avoid using the type-param anyways (for roughly the same reasons that you shouldn't assume `F === typeof(F).instance`):

```julia
struct TypeArgFunction{n, F, T} <: Function
    f::F
    t::Type{T}
    TypeArgFunction{n}(f, t) = new{Core.Typeof(f), Type{t}}(f, t)
end
(f::TypeArgFunction{1})(args...) = f.f(f.t, args...)
(f::TypeArgFunction{2})(arg1, args...) = f.f(arg1, f.t, args...)
# TODO: function (f::TypeArgFunction{N})(args...) where {N}
#    preargs, postargs = split(args, N)
#    return f.f(preargs..., f.t, postargs...)
# end
```

Reply:
Or it might be interesting to note that this is just a hand-coded special case of the `_capturescalars` function (of which `make_makeargs` is also just a copy), and use that function instead.